executor/benchmark_test.go

```diff
@@ -0,0 +1,411 @@
+		if aggDesc.HasDistinct {
+			e.isUnparallelExec = true
+		}
+	}
+	if finalCon, partialCon := sessionVars.HashAggFinalConcurrency, sessionVars.HashAggPartialConcurrency; finalCon <= 0 || partialCon <= 0 || finalCon == 1 && partialCon == 1 {
+		e.isUnparallelExec = true
+	}
+	partialOrdinal := 0
+	for i, aggDesc := range v.aggFuncs {
+		if e.isUnparallelExec {
+			e.PartialAggFuncs = append(e.PartialAggFuncs, aggfuncs.Build(v.ctx, aggDesc, i))
+		} else {
+			ordinal := []int{partialOrdinal}
+			partialOrdinal++
+			if aggDesc.Name == ast.AggFuncAvg {
+				ordinal = append(ordinal, partialOrdinal+1)
+				partialOrdinal++
+			}
+			partialAggDesc, finalDesc := aggDesc.Split(ordinal)
+			partialAggFunc := aggfuncs.Build(v.ctx, partialAggDesc, i)
+			finalAggFunc := aggfuncs.Build(v.ctx, finalDesc, i)
+			e.PartialAggFuncs = append(e.PartialAggFuncs, partialAggFunc)
+			e.FinalAggFuncs = append(e.FinalAggFuncs, finalAggFunc)
+			if partialAggDesc.Name == ast.AggFuncGroupConcat {
+				finalAggFunc.(interface{ SetTruncated(t *int32) }).SetTruncated(
+					partialAggFunc.(interface{ GetTruncated() *int32 }).GetTruncated(),
+				)
+			}
+		}
+		if e.defaultVal != nil {
+			value := aggDesc.GetDefaultValue()
+			e.defaultVal.AppendDatum(i, &value)
+		}
+	}
+	return e
+}
+
+func buildStreamAggExecutor(v *aggExecutorParameters) Executor {
+	e := &StreamAggExec{
+		baseExecutor: newBaseExecutor(v.ctx, v.schema, "", v.child),
+		StmtCtx:      v.ctx.GetSessionVars().StmtCtx,
+		aggFuncs:     make([]aggfuncs.AggFunc, 0, len(v.aggFuncs)),
+		GroupByItems: v.groupByItems,
+	}
+	if len(v.groupByItems) != 0 || aggregation.IsAllFirstRow(v.aggFuncs) {
+		e.defaultVal = nil
+	} else {
+		e.defaultVal = chunk.NewChunkWithCapacity(e.retTypes(), 1)
+	}
+	for i, aggDesc := range v.aggFuncs {
+		aggFunc := aggfuncs.Build(v.ctx, aggDesc, i)
+		e.aggFuncs = append(e.aggFuncs, aggFunc)
+		if e.defaultVal != nil {
+			value := aggDesc.GetDefaultValue()
+			e.defaultVal.AppendDatum(i, &value)
+		}
+	}
+
+	return e
+}
+
+type aggTestCase struct {
+	// The test table's schema is fixed (aggCol Double, groupBy Long).
+	exec        string // "hash" or "stream"
+	aggFunc     string // sum, avg, count ....
+	groupByNDV  int    // the number of distinct group-by keys
+	hasDistinct bool
+	rows        int
+	concurrency int
+}
+
+func (a aggTestCase) String() string {
+	return fmt.Sprintf("(%v|%v|%v|%v|%v|%v)",
+		a.exec, a.aggFunc, a.hasDistinct, a.rows, a.groupByNDV, a.concurrency)
+}
+
+func defaultAggTestCase(exec string) *aggTestCase {
+	return &aggTestCase{exec, ast.AggFuncSum, 1000, false, 10000000, 4}
+}
+
+func buildAggDataSource(b *testing.B, cas *aggTestCase) *mockDataSource {
+	fieldTypes := []*types.FieldType{
+		types.NewFieldType(mysql.TypeDouble), types.NewFieldType(mysql.TypeLong),
+	}
+	cols := make([]*expression.Column, len(fieldTypes))
+	for i := range cols {
+		cols[i] = &expression.Column{Index: i, RetType: fieldTypes[i]}
+	}
+	ctx := core.MockContext()
```

Comment:
`mock.NewContext()` is enough?

We'd better follow this comment `// MockContext is only used for plan related tests.` or add one in `mock` package.

Reply:
Resolved by using `util/mock.ConText` to replace it.